typescript/sdk/src/token/EvmERC20WarpModule.ts

```diff
@@ -148,46 +152,84 @@ export class EvmERC20WarpModule extends HyperlaneModule<
     assert(actualConfig.remoteRouters, 'actualRemoteRouters is undefined');
     assert(expectedConfig.remoteRouters, 'actualRemoteRouters is undefined');
 
-    // We normalize the addresses for comparison
-    actualConfig.remoteRouters = Object.fromEntries(
-      Object.entries(actualConfig.remoteRouters).map(([key, value]) => [
-        key,
-        // normalizeConfig removes the address property but we don't want to lose that info
-        { ...normalizeConfig(value), address: normalizeConfig(value.address) },
-      ]),
+    const { remoteRouters: actualRemoteRouters } = actualConfig;
+    const { remoteRouters: expectedRemoteRouters } = expectedConfig;
+
+    const routesToEnroll = Array.from(
+      difference(
+        new Set(Object.keys(expectedRemoteRouters)),
+        new Set(Object.keys(actualRemoteRouters)),
+      ),
     );
-    expectedConfig.remoteRouters = Object.fromEntries(
-      Object.entries(expectedConfig.remoteRouters).map(([key, value]) => [
-        key,
-        // normalizeConfig removes the address property but we don't want to lose that info
-        { ...normalizeConfig(value), address: normalizeConfig(value.address) },
-      ]),
+
+    if (routesToEnroll.length === 0) {
+      return updateTransactions;
+    }
+
+    const contractToUpdate = TokenRouter__factory.connect(
+      this.args.addresses.deployedTokenRoute,
+      this.multiProvider.getProvider(this.domainId),
     );
 
+    updateTransactions.push({
+      chainId: this.chainId,
+      annotation: `Enrolling Router ${this.args.addresses.deployedTokenRoute} on ${this.args.chain}`,
+      to: contractToUpdate.address,
+      data: contractToUpdate.interface.encodeFunctionData(
+        'enrollRemoteRouters',
+        [
+          routesToEnroll.map((k) => Number(k)),
+          routesToEnroll.map((a) =>
+            addressToBytes32(expectedRemoteRouters[a].address),
+          ),
+        ],
+      ),
+    });
+
+    return updateTransactions;
+  }
```

Comment:
Shouldn't we also add txs to remove/add the destination gas (in another method)?

Reply:
that should be triggered by update to config.destinationGas. functionally, unenrolling will also "remove" the destinationGas from warp read.